### ISSUE TITLE
Write a terms of service and privacy policy

### ISSUE THREAD

**Issue by Gargron** [tootsuite/mastodon#115]
I've started on it using a generic ToS generator, current page can be viewed here: <https://mastodon.social/terms>, however, some points from the generated text are a really bad fit for Mastodon, such as the limit on replication of information on other servers (literally what Mastodon is about!) as well as all the copyright stuff (it's AGPLv3 for the software, I suppose users must have some form of license on their own content by default too, and it might be wise to write in a Creative Commons attribution license on that to prevent trouble). It's also lacking points about rules of conduct/content and termination of service.

Somebody who is good at lawyering please help me

**Comment by donkeyontheedge** [comment]
Interesting challenge. I had a look at the cc licenses and they don't really fit the project?

I'll go have a look at your terms (not a lawyer btw) but suggest might be an idea to make a bullet list of what you want?

1. Project objectives

  a) Promote the growth of the product network
  b) Promote the user adoption of a global, social network not dependent on one vendor
  c) Provide the key community infrastructure under threat by the collapse of Twitter

2. Use of the software: do / don't do? assume federation will mean classic open source/gnu problems with clarity and organisation but presumably if your design creates a clear way in to the service this shouldn't be too much of a problem? Also wonder how much SLA-type stuff you need about performance, server security etc. Maybe none and we assume common sense?
3. Code of conduct (see example http://thingmonk.com/coc.html) 
4. Other stuff

**Comment by donkeyontheedge** [comment]
Hi Gargron I had a look at you page at https://mastodon.social/terms it looks ok but I would suggest that getting lost in the detail of how people can use the software might not be the best approach. I guess you need advice from a heavy-hitter like @aral but in my opinion this is a give-to-get experience (the driving force of the whole damn internet before marketing basterds got involved) and we have a very real and immediate need to keep Twitter's community infrastructure in place as it melts down.

It's a big complicated topic and I guess this needs to be a do-ocracy but I would say the best starting point is give it away with no restrictions. Create restrictions at the platform use level

- block hate accounts
- black-list servers that are set up with evil config
- other ideas?

tl;dr: Start as open as possible

Cheers,
Dug

**Comment by aral** [comment]
Mastodon is decentralised, which means that terms for centralised services will not apply verbatim. In fact, I would suggest starting from scratch.

For one thing, it doesn’t make sense to have terms for Mastodon in general. The beauty of decentralised systems is that everyone gets to run their own home where their word is law. So mastodon.social can have its terms (including how it deals with harassment, etc.) and indie.social can have its own terms and <your-name-here>.social can have its own terms. Of course then mastodon.social is free to block other servers, etc., (which, as I understand it, it already does). 

Instead of starting with a centralised mindset, let’s take a moment to think about what curation means for a decentralised system.

Let’s start with an empty canvas and design :)

(And I agree with a lot of Dug’s point, above, including his conclusions. Will try to help as much as I can.)

**Comment by Gargron** [comment]
I think there are two parts to this. I think some part of terms of service/privacy policy depends solely on software, and not on decisions of the person running it. And then there are parts that can be customized depending on the admin's preference... But there needs to be a sensible default, because nobody wants to have to write a legal document when they want to try out hosting a new instance. (I know I don't! XP)

Starting with a blank slate sounds appealing but only because then I don't have to write anything. But people were asking for a privacy policy today, so I think it's an important factor. (I hear that while terms of service are not legally required to be present, a privacy policy needs to exist)

**Comment by jimmytidey** [comment]
Wow this is really complicated. Is it right that all these different things are being discussed here?

1. The licence for the Mastodon code itself 
2. A default set of T&Cs which might be useful to someone setting up an instance
3. The terms on which you can federate with the existing instance at mastodon.social 
4. The actual set of T&Cs for mastodon.social itself
5. An overall set of principles behind the project

On 1, it sounds like the answer is AGPL v3. 

On 2, 3 and 4 it seems that they depend on 5. 

Would it be appropriate to start a new Github issue just for 5? I think that would be a really useful space to encourage new people to join the community, and also play a part in its principles. 

Once we have that perhaps we can find some kind of expert to help craft the exact wording.

**Comment by jimmytidey** [comment]
Some thoughts I have, some of which may be more or less achievable. 

-To provide a federated service not controlled by any one institution or individual 

- To provide a clear and simple UX, even for users who do not understand federation

- To provide a service where searching and conversations are seamless regardless of the underlying architecture

- To provide freedom of speech while providing excellent tools for users to protect themselves 

- Mastodon social is a response to the current commercial uncertainties around Twitter but doesn't aspire to be a clone

- To provide software with an open license so that anyone else can set up their own federated network 

- To achieve sustainability without advertising (through voluntary contributions?)

- To protect users private data 

- To make public data available through APIs and portable, and to encourage a third party ecosystem

**Comment by Gargron** [comment]
Perhaps I could borrow something from the Discourse ones? https://github.com/discourse/discourse/blob/f0571722161c807a9c38376d2285db962102dc99/config/locales/server.en.yml#L2740 End goal would be making them dynamically editable by the instance admin, but we need a sensible default.

**Comment by joshtriplett** [comment]
@Gargron Might I suggest talking to the folks at Conservancy (https://sfconservancy.org/), and asking them for some legal advice that fits this particular use case? They can help with appropriate terms of service that actually fit what you want rather than just boilerplate.

**Comment by Gargron** [comment]
I have copied the privacy policy from the Discourse project because it fits quite perfectly without modifications. The Discourse TOS is a bit muddier, since they offer paid services, enterprise upgrades, ads...

@joshtriplett Do I just send an e-mail to someone from their contacts page?

**Comment by joshtriplett** [comment]
@Gargron Try info@sfconservancy.org , or drop by their IRC channel. https://sfconservancy.org/about/contact/

**Comment by TazeTSchnitzel** [comment]
I believe having a proper privacy policy is also needed for Data Protection Directive compliance, which matters for people running instances in the EU (like @Gargron is). Actually, with the new regulations, I think it'll eventually matter even for non-EU people if you have EU users.

**Comment by ineffyble** [comment]
New page RE: content moderation is relevant - https://mastodon.social/about/more

**Comment by Gargron** [comment]
From my research, "terms of service" is an optional thing, while "privacy policy" is not. Having copied the privacy policy from the Discourse project (which copied it from WordPress afaik), and having added the ability for admins to write custom about page content (and using it to add guidelines on mastodon.social), I think this particular issue can be closed as resolved.
